### PR TITLE
add wigxjpf

### PR DESCRIPTION
**Pull request by 0382**
The wigxjpf library is used for evaluating Wigner symbols, which is very useful for angular momentum coupling. The GNU-GSL library offers `gsl_sf_coupling_?j` functions, but they are inaccurate for very large angular momentum. This library use big integer techniques to give accurate result.

### CODE REVIEW

**Comment by 0382** [comment]
I have changed the script following the suggestions.

File: packages/w/wigxjpf/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    set_homepage("https://fy.chalmers.se/subatom/wigxjpf/")
+    set_description("WIGXJPF evaluates Wigner 3j, 6j and 9j symbols accurately using prime factorisation and multi-word integer arithmetic.")
+    set_license("GPL-3.0", "LGPL-3.0")
+
+    set_urls("https://fy.chalmers.se/subatom/wigxjpf/wigxjpf-$(version).tar.gz")
+    add_versions("1.13", "90ab9bfd495978ad1fdcbb436e274d6f4586184ae290b99920e5c978d64b3e6a")
+
+    on_install("linux", "macosx", function (package)
+        local configs = {}
+        import("package.tools.make").build(package, configs)
```
Comment (waruqi) [inline packages/w/wigxjpf/xmake.lua:11]: We need to distinguish between static/shared libraries based on `package:config("shared")` if it only provides static libraries, please mark it as static readonly. 

```
add_configs("shared",      {description = "Build shared library.", default = false, type = "boolean", readonly = true})
```

File: packages/w/wigxjpf/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        os.cp("inc/*.h", package:installdir("include"))
+        os.cp("lib/*.a", package:installdir("lib"))
+        os.cp("bin/*", package:installdir("bin"))
+    end)
+
+    on_test(function (package)
+        assert(package:check_csnippets({test = [[
```
Comment (star-hengxing) [inline packages/w/wigxjpf/xmake.lua:18]: We can simplify test with [has_cfuncs](https://xmake.io/#/manual/package_instance?id=packagehas_cfuncs)

File: packages/w/wigxjpf/xmake.lua (hunk 1 above)
Comment (0382) [reply packages/w/wigxjpf/xmake.lua:11]: In the Makefile, the shared library is maked by a extra target, thus by running `make shared`. I add some code for the condition.

File: packages/w/wigxjpf/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    set_license("GPL-3.0", "LGPL-3.0")
+
+    set_urls("https://fy.chalmers.se/subatom/wigxjpf/wigxjpf-$(version).tar.gz")
+    add_versions("1.13", "90ab9bfd495978ad1fdcbb436e274d6f4586184ae290b99920e5c978d64b3e6a")
+
+    on_install("linux", "macosx", function (package)
+        local configs = {}
+        import("package.tools.make").build(package, configs)
+        os.cp("inc/*.h", package:installdir("include"))
+        os.cp("lib/*.a", package:installdir("lib"))
```
Comment (waruqi) [inline packages/w/wigxjpf/xmake.lua:13]: if the current package is shared library, we cannot install static lib files.

Comment (0382) [reply packages/w/wigxjpf/xmake.lua]: In the Makefile, the shared library depend on the static library. So I think I can just do not copy the static library when installing shared library.

**Pull request merged**